homeassistant/components/influxdb.py

```diff
@@ -77,6 +77,11 @@ def influx_event_listener(event):
             _state = 0
         else:
             _state = state.state
+            if _state and isinstance(_state, str):
```

Comment:
`_state` will always be defined and will always be a string. Python strings have a `isnumeric` method, so using that would be best:
```python
if _state.isnumeric():
  _state = float(_state)
```